File: tests/src/main/java/com/github/klee0kai/test/tech/components/DataStorage.java
package com.github.klee0kai.test.tech.components;

import com.github.klee0kai.test.di.base_phone.identifiers.DataStorageSize;

import java.util.UUID;

public class DataStorage {

    public final UUID uuid = UUID.randomUUID();

    public final String size;

    public DataStorage() {
        size = null;
    }

    public DataStorage(DataStorageSize size) {
        this.size = size.size;
    }

}

File: tests/src/main/java/com/github/klee0kai/test/tech/components/DataStorage.kt
package com.github.klee0kai.test.tech.components

import com.github.klee0kai.test.di.base_phone.identifiers.DataStorageSize
import java.util.*

class DataStorage {
    @JvmField
    val uuid: UUID = UUID.randomUUID()

    @JvmField
    val size: String?

    constructor() {
        size = null
    }

    constructor(size: DataStorageSize?) {
        this.size = size?.size
    }
}
